### DISCUSSION TITLE
Tausender Trennzeichen und Nachkommastellen

### DISCUSSION THREAD

**Discussion by tschortschde** [Zefau/ioBroker.jarvis#2394] [Q&A] [answered]
Wer kann mir bitte sagen, wie der Code für die Datenpunkt Eigenschaften des Gerätes lautet, wenn man das Tausender-Trennzeichen und keine Nachkommastellen für einen Wert haben möchte? 
Ich suche speziell für die Helligkeitsanzeige: Der Wert "2000.75" soll gerundet folgendes Format haben: "2.001"

**Comment by mcuiobroker** [comment]
Geht momentan nicht, da Zefau anscheinend eine Funktion eingebaut hat, die auch Srtingwerte wieder in normales Format zurücksetzt.
Würde nur als JSON-Ausgabe gehen:
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/a142cb18-b168-43aa-879a-36d2f41305b4)
oder so
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/d6974530-18ad-4e21-8345-c0afa09841af)

**Reply by tschortschde** [reply]
Ist es möglich, den Wert zumindest auf mit Tausendertrennzeichen darzustellen und die Nachkommastellen zu kürzen?
Wie wäre dann der Code?

**Reply by mcuiobroker** [reply]
So?
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/0597f660-588d-4b13-a280-75e1f5f0239f)
```
// In die Anzeigevariante 
{"default":"(Number(parseFloat({val}).toFixed(1)).toLocaleString('de-DE'))"}
```

**Comment by tschortschde** [comment]
Habe es genau so gemacht 
{"default":"(Number(parseFloat({val}).toFixed(**0**)).toLocaleString('de-DE'))"}
und bekomme leider ein Komma statt einem Punkt ;-(
![image](https://github.com/Zefau/ioBroker.jarvis/assets/11274233/1c864c58-3f09-45d9-9298-342722b0793f)
Geht das nur mit einer Nachkommastelle?

Wie hast Du es denn so hinbekommen?
![image](https://github.com/Zefau/ioBroker.jarvis/assets/11274233/86055524-d263-4fab-a696-82fcd18d6d35)

**Reply by mcuiobroker** [reply, marked as answer]
toFixed(0) geht nicht nimm toFixed(1)

Andere Varainte über Datenpunkt-Eigenschaften:
```
{"value":"val => {const fmtDe = Intl.NumberFormat('de-DE'); let aval = fmtDe.format(Math.round(val));  if(aval.indexOf(',')>0){return (JSON.stringify(aval)).substring(1,aval.length)}else{return (JSON.stringify(aval)).substring(1,)}}"}
```

**Comment by tschortschde** [comment]
Danke. Woher kommt den das Anführungszeichen? 
Das könnte man doch sicher ausblenden, oder  ;-)

**Reply by mcuiobroker** [reply]
Nein, dann steht dort wieder 2,001.
Anführungszeichen kommt vom JSON.stringify.
